mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/common/SearchHelper.java

```diff
@@ -0,0 +1,189 @@
+package org.mifosplatform.integrationtests.common;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.path.json.JsonPath.from;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.junit.Test;
+
+import com.jayway.restassured.specification.RequestSpecification;
+import com.jayway.restassured.specification.ResponseSpecification;
+
+public class SearchHelper {
+	
+	private final RequestSpecification requestSpec;
+    private final ResponseSpecification responseSpec;
+    
+    private static final String SEARCH_CLIENT_URL = "/mifosng-provider/api/v1/search?"+ Utils.TENANT_IDENTIFIER;
+    private static final String GET_CLIENT_URL = "/mifosng-provider/api/v1/clients/"  + Utils.TENANT_IDENTIFIER;
+    static HashMap<String, Object> detail=new HashMap<String,Object>(); 
+    
+    public SearchHelper(final RequestSpecification requestSpec, final ResponseSpecification responseSpec) {
+        this.requestSpec = requestSpec;
+        this.responseSpec = responseSpec;
+    }
+   
+    public static Object exactClientSearch (final RequestSpecification requestSpec, final ResponseSpecification responseSpec, final String searchParameter,
+            final String jsonReturn) {
+        final String SEARCH_CLIENT_URL = "/mifosng-provider/api/v1/search?exactMatch="+true+"&query="+searchParameter+"&resource="+"clients,clientIdentifiers"+"&"+ Utils.TENANT_IDENTIFIER;       
+        return Utils.performServerGet(requestSpec, responseSpec, SEARCH_CLIENT_URL, jsonReturn);
+    }     
+    
+    public static ArrayList<HashMap> exactMatchSearchClient(final RequestSpecification requestSpec, final ResponseSpecification responseSpec,
+    final String searchParameter){
+    	    	
+   	ArrayList<HashMap> searchResults = (ArrayList<HashMap>) exactClientSearch(requestSpec, responseSpec, searchParameter, "" );
+    return searchResults;	   
+   }
+    public static Object likeClientSearch (final RequestSpecification requestSpec, final ResponseSpecification responseSpec, final String searchParameter,
+            final String jsonReturn) {
+        final String SEARCH_CLIENT_URL = "/mifosng-provider/api/v1/search?exactMatch="+false+"&query="+searchParameter+"&resource="+"clients,clientIdentifiers"+"&"+ Utils.TENANT_IDENTIFIER;       
+        return Utils.performServerGet(requestSpec, responseSpec, SEARCH_CLIENT_URL, jsonReturn);
+    }  
+    
+    public static ArrayList<HashMap> likeMatchSearchClient(final RequestSpecification requestSpec, final ResponseSpecification responseSpec,
```

Comment:
Let the helper functions have return type as Object.Type cast accordingly in the calling method based on the data we require.